File: IsPalindrome.swift
import Foundation

func isPalindrome(_ s:String)->Bool{
    let characters = Array(s)
    var left = 0
    var right  = characters.count-1

    while left<right{
        if characters[left] != characters[right]{
            return false
        }
        left += 1
        right -= 1
    }
    return true

}

let s = readLine()!
if isPalindrome(s){
    print("Yes")
}else{
    print("No")
}

File: IsPalindrome.cpp
#include<bits/stdc++.h>
using namespace std;

bool isPalindrome(string s){
    int left = 0, right = s.length()-1;
    while(left<right){
        if(s[left] != s[right])
        return false;
        left++;
        right--;
    }
    return true;
}

int main()
{
string s;
cin>>s;
cout<<(isPalindrome(s)? "Yes" : "No")<<endl;

    return 0;
}